tls/s2n_signature_algorithms.c

```diff
@@ -115,6 +115,20 @@ int s2n_get_and_validate_negotiated_signature_scheme(struct s2n_connection *conn
         }
     }
 
+    /* We require an exact match in TLS 1.3, but all previous versions can fall back to the default SignatureScheme.
+     * This means that an s2n client will accept the default SignatureScheme from a TLS server, even if the client did
+     * not send it in it's ClientHello. This pre-TLS1.3 behavior is an intentional choice to maximize support. */
+    struct s2n_signature_scheme default_scheme;
+    GUARD(s2n_choose_default_sig_scheme(conn, &default_scheme));
```

Comment:
Nit: Since we are setting the default signature scheme only for <= TLS1.2 can we move this portion of the code within the if condition on line 124, as [s2n_choose_default_sig_scheme](https://github.com/awslabs/s2n/blob/master/tls/s2n_signature_algorithms.c#L142) sets the signature scheme for TLS1.3 as well.

Reply:
I don't think `s2n_choose_default_sig_scheme()` changes anything other than the `default_scheme` variable that's pass into it.